entrypoint/ocp/build_secrets.go

```diff
@@ -0,0 +1,98 @@
+package ocp
+
+/*
+	Handle Secrets provided by OpenShfit.
+*/
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// copySecret is a hack. OpenShift presents secrets with 0400.
+// Since syscall.SetUid is blocked, we need to use sudo read the file.
+func copySecret(inDir, name, from string) error {
+	sDir := filepath.Join(inDir, "secrets", name)
+	if err := os.MkdirAll(sDir, 0755); err != nil {
+		return err
+	}
+
+	baseName := filepath.Base(from)
+	to := filepath.Join(sDir, baseName)
+	if _, err := os.Stat(to); err == nil {
+		log.Infof("Already handled secret %s", to)
+		return nil
+	}
+
+	cmd := exec.Command("sudo", "cat", from)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return err
+	}
+	// OCP secrets are limited at 1Mb....so this "safe-ish"
+	if err := ioutil.WriteFile(to, out, 0555); err != nil {
+		return err
+	}
+	log.Infof("Created secret: %s", to)
+
+	// If an envVar SECRET_MAP_FILE_$SECRET matches split the value
+	// and set left side to the right.
+	// i.e. SECRET_MAP_FILE_MY_SECRET with "AWS_SECRET=CONFIG",
+	// then "AWS_SECRET" would be set the <PATH>/CONFIG
+	e := fmt.Sprintf("%s%s", cosaSecretMapFile, name)
+	ev, found := os.LookupEnv(e)
+	if found {
+		v := strings.Split(ev, "=")
+		if len(v) != 2 {
+			return fmt.Errorf("%s envVar should be in format 'K=V', not %s", e, ev)
+		}
+		if v[1] == baseName {
+			os.Setenv(v[0], to)
+			log.Debugf("Set envVar: %s=%s", v[0], to)
+		}
+	}
+
+	return nil
+}
+
+// buildSecretsSetup copies the pod mapped secrets so that the build-user is
+// able to read them. Secrets on 3.x clusters are mapped to 0400 and owned by root.
```

Comment:
Hmm wouldn't that be a function of the user the build runs as?  Are these builds running as uid 0 or part of `runAsRange`?

Reply:
On 3.x, there run as `0` which COSA then switches to `builder`. So `entry` runs as root since it runs before the `cosa` commands.

Reply:
Is that still true in 4.x?  Wouldn't we want to match what the current FCOS pipeline does and `runAsAny` i.e. use a dynamic UID - I don't think we use `sudo` at all there.

Reply:
You are correct that the behavior for 4.x is different. Truthfully, I would hope that this goes away once we have all the builds on 4.x clusters.  I'm filing a PR to get the permissions to do builds on `api.ci` (and figure out how to get the context from behind the firewall).